Gost yescrypt may fail to build on some older versions of glibc.

**Issue Description**
```
/home/william/development/389ds/ds/ldap/servers/plugins/pwdstorage/gost_yescrypt.c:23:12: warning: implicit declaration of function 'crypt_rn' is invalid in C99 [-Wimplicit-function-declaration]
    hash = crypt_rn(userpwd, dbpwd, &output, (int) sizeof(output));
           ^
/home/william/development/389ds/ds/ldap/servers/plugins/pwdstorage/gost_yescrypt.c:23:10: warning: incompatible integer to pointer conversion assigning to 'char *' from 'int' [-Wint-conversion]
    hash = crypt_rn(userpwd, dbpwd, &output, (int) sizeof(output));
         ^ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/william/development/389ds/ds/ldap/servers/plugins/pwdstorage/gost_yescrypt.c:41:15: error: use of undeclared identifier 'CRYPT_GENSALT_OUTPUT_SIZE'
    char salt[CRYPT_GENSALT_OUTPUT_SIZE];
              ^
/home/william/development/389ds/ds/ldap/servers/plugins/pwdstorage/gost_yescrypt.c:47:10: warning: implicit declaration of function 'crypt_gensalt_rn' is invalid in C99 [-Wimplicit-function-declaration]
    if (!crypt_gensalt_rn(prefix, 0, ((void*)0), 0, salt, (int) sizeof(salt))) {
         ^
/home/william/development/389ds/ds/ldap/servers/plugins/pwdstorage/gost_yescrypt.c:53:12: warning: implicit declaration of function 'crypt_rn' is invalid in C99 [-Wimplicit-function-declaration]
    hash = crypt_rn(pwd, salt, &output, (int) sizeof(output));
           ^
4 warnings and 1 error generated.
```

This appears to be due to missing defines in crypt.h on older glibc. 

**Package Version and Platform:**
SLE 15.2 glibc-2.26-13.51.1.x86_64

does not affect SUSE TW glibc-2.32-4.1.x86_64

--- a/ldap/servers/plugins/pwdstorage/gost_yescrypt.c
+++ b/ldap/servers/plugins/pwdstorage/gost_yescrypt.c
@@ -7,11 +7,12 @@
 #include <config.h>
 #endif
 
-#include <crypt.h>
-#include <errno.h>
-
 #include "pwdstorage.h"
 
+#include <crypt.h>
+
+#ifdef XCRYPT_VERSION_STR
+#include <errno.h>
 int
 gost_yescrypt_pw_cmp(const char *userpwd, const char *dbpwd)
 {
@@ -62,3 +63,25 @@ gost_yescrypt_pw_enc(const char *pwd)
 
     return enc;
 }
+
+#else
+
+/*
+ * We do not have xcrypt, so always fail all checks.
+ */
+int
+gost_yescrypt_pw_cmp(const char *userpwd __attribute__((unused)), const char *dbpwd __attribute__((unused)))
+{
+    slapi_log_err(SLAPI_LOG_ERR, GOST_YESCRYPT_SCHEME_NAME,
+                  "Unable to use gost_yescrypt_pw_cmp, xcrypt is not available.\n");
+    return 1;
+}
+
+char *
+gost_yescrypt_pw_enc(const char *pwd __attribute__((unused)))
+{
+    slapi_log_err(SLAPI_LOG_ERR, GOST_YESCRYPT_SCHEME_NAME,
+                  "Unable to use gost_yescrypt_pw_enc, xcrypt is not available.\n");
+    return NULL;
+}
+#endif

